src/praisonai/praisonai/bots/_delivery.py

```
@@ -342,12 +351,23 @@ async def deliver_with_retry(
+            # Honour a server-mandated wait (Telegram retry_after, HTTP
+            # Retry-After) over the generic backoff estimate.
+            mandated = server_retry_after(e)
+            delay = mandated if mandated is not None else compute_backoff(backoff, attempt)
             
+            # Widen this channel's lane so the next send does not immediately
+            # re-trip the platform limit.
+            if rate_limiter is not None and delay > 0:
+                try:
+                    await rate_limiter.penalise(channel_id, delay)
+                except Exception:
+                    pass
```

Ruff (0.15.18)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)